ingestion-sink/src/main/java/com/mozilla/telemetry/ingestion/sink/config/SinkConfig.java

```diff
@@ -72,31 +84,58 @@ Output getOutput(Env env) {
+            env.getLong(BATCH_MAX_BYTES, 100_000_000L), // default 100MB
+            env.getInt(BATCH_MAX_MESSAGES, 1_000_000), // default 1M messages
+            env.getDuration(BATCH_MAX_DELAY, "10m"), // default 10 minutes
+            PubsubMessageToTemplatedString.of(getGcsOutputBucket(env)), getFormat(env),
+            ignore -> CompletableFuture.completedFuture(null)));
+      }
 
-        // if OUTPUT_TOPIC is present send notification for each gcs blob written
-        final Function<BlobInfo, CompletableFuture<Void>> batchCloseHook;
-        if (env.containsKey(OUTPUT_TOPIC)) {
-          Function<PubsubMessage, CompletableFuture<Void>> pubsubWrite = pubsub.getOutput(env);
-          batchCloseHook = blobInfo -> pubsubWrite.apply(BlobInfoToPubsubMessage.apply(blobInfo));
-        } else {
-          batchCloseHook = ignore -> CompletableFuture.completedFuture(null);
-        }
+      @Override
+      long getDefaultMaxOutstandingElementCount() {
+        return 1_000_000L; // 1M messages
+      }
+
+      @Override
+      long getDefaultMaxOutstandingRequestBytes() {
+        return 1_000_000_000L; // 1GB
+      }
+    },
+    bigQueryLoad {
+
+      @Override
+      Output getOutput(Env env) {
+        return new Output(env, this, new BigQuery.Load(getBigQueryService(env), getGcsService(env),
+            // BigQuery Load API limits maximum bytes per request to 15TB
+            env.getLong(LOAD_MAX_BYTES, 15_000_000_000_000L), // default 15TB
+            // BigQuery Load API limits maximum files per request to 10,000
+            env.getInt(LOAD_MAX_FILES, 10_000), // default 10K files
+            // BigQuery Load API limits maximum load requests per table per day to 1,000 so with
+            // three instances for redundancy that means each instance should load at most once
+            // every 259.2 seconds or approximately 4.3 minutes
+            env.getDuration(LOAD_MAX_DELAY, "10m"), // default 10 minutes
+            BigQuery.Load.Delete.onSuccess)); // don't delete files until successfully loaded
+      }
+    },
+    bigQueryFiles {
 
-        return new Output(env, this,
-            new Gcs.Write.Ndjson(StorageOptions.getDefaultInstance().getService(),
-                env.getLong(BATCH_MAX_BYTES, 100_000_000L), // default 100MB
-                env.getInt(BATCH_MAX_MESSAGES, 1_000_000), // default 1M messages
-                env.getDuration(BATCH_MAX_DELAY, "10m"), // default 10 minutes
-                PubsubMessageToTemplatedString.of(gcsPrefix), getFormat(env), batchCloseHook));
+      @Override
+      Output getOutput(Env env) {
+        Output pubsubWrite = pubsub.getOutput(env);
+        return new Output(env, this, new Gcs.Write.Ndjson(getGcsService(env),
+            // BigQuery Load API limits maximum bytes per file to 4GB compressed or 5TB uncompressed
+            env.getLong(BATCH_MAX_BYTES, 100_000_000L), // default 100MB
```

Comment:
This value seems low compared to what's discussed the comment.

Reply:
Yeah, the default value wasn't chosen based on the limit, it was just copied from the gcs output. I'll add a justification of the number, which we can update